### ISSUE TITLE
Azure preset: Make staticwebapp.config.json optional

### ISSUE THREAD

**Issue by jegardiner** [nitrojs/nitro#1087]
### Describe the feature

The Azure preset currently regenerates the staticwebapp.config.json file on build. Overwriting any additional configuration that you may have added to this file, such as route restrictions using Azure's built-in authentication.

[You can see the code in question here.](https://github.com/unjs/nitro/blob/main/src/presets/azure.ts#L134-L136)

For example, here is a staticwebapp.config.json _before_ `npm run build`:

```
{
  "platform": {
    "apiRuntime": "node:16"
  },
  "routes": [
    {
      "route": "/index.html",
      "redirect": "/"
    },
    {
      "route": "/",
      "rewrite": "/api/server"
    },
    {
      "route": "/admin",
      "allowedRoles": ["authenticated"]
    },
    {
      "route": "/logout",
      "redirect": "/.auth/logout"
    },
  ],
  "responseOverrides": {
    "401": {
      "statusCode": 302,
      "redirect": "/.auth/login/aad"
    }
  },
  "navigationFallback": {
    "rewrite": "/api/server"
  }
}
```

And the same file _after_ `npm run build`:

```
{
  "platform": {
    "apiRuntime": "node:16"
  },
  "routes": [
    {
      "route": "/index.html",
      "redirect": "/"
    },
    {
      "route": "/",
      "rewrite": "/api/server"
    }
  ],
  "navigationFallback": {
    "rewrite": "/api/server"
  }
}
```

Unless there is already a way to add this supplemental configuration so that Nitro will include it, would it be possible to add a configuration option to toggle whether the staticwebapp.config.json should be generated during build.

Until then, I'll be using a hook to replace the file after the build completes.

### Additional information

- [X] Would you be willing to help implement this feature?

**Comment by tobiasdiez** [comment]
Related: https://github.com/unjs/nitro/discussions/420

**Comment by danielroe** [comment]
I agree. We currently merge Netlify presets and I think this would be doable here too...

**Comment by tobiasdiez** [comment]
@danielroe do you have a vision on how modules can provide deployment-preset-dependent config? For example, in order to set the correct `x-robots-tag` header the nuxt-simple-router module would need to set the [headers](https://learn.microsoft.com/en-us/azure/static-web-apps/configuration#define-routes) option for each route. For netlify, it would be the `_headers` file. In this particular case, one could probably find a workaround by setting `route.headers` in the module. But there are a few other configs that would really need to be set by the module, e.g. trailing slash in [nuxt-seo-kit](https://github.com/harlan-zw/nuxt-seo-kit/#enabling-trailing-slash) setting it in the [azure config](https://learn.microsoft.com/en-us/azure/static-web-apps/configuration#trailing-slash).

---

If this is too off-topic, I'll create new issue.

**Comment by danielroe** [comment]
@tobiasdiez I would set a headers routeRule. Then Nitro will be responsible for ensuring the appropriate preset setting.

**Comment by harlan-zw** [comment]
Slightly off-topic but since @tobiasdiez pinged me in the linked issue:

As far as I can tell route rule headers don't get applied through h3 for public static files, at least in dev and `node-server`. (reproduction: https://stackblitz.com/edit/nuxt-starter-jobvwh?file=nuxt.config.ts)

The behavior would ideally be consistent across presets. Should I make a separate issue @danielroe or am I missing something?

**Comment by calbers1** [comment]
Myself and the rest of the dev team have been banging our heads against a wall over this for the past couple of days. Is there any workaround for the staticwebapp.config.json file changes being automatically wiped out? We reeeeally need to make some edits to it for some routing issues caused by legacy software (we aren't able to change the endpoint being hit by the software, which isn't in the api folder, and it requires POST to work).

**Comment by danielroe** [comment]
I'd welcome a PR on this. We obviously need to allow overwriting quite a lot of the file, but if you can propose a good heuristic we can probably implement quite quickly.

**Comment by NicholasDawson** [comment]
I'm running into the same issue and I can start work on a PR for this issue. I have included my proposed solution below, I would appreciate feedback and/or approval to begin work on this. I have no problem doing the work as long as I know the solution is generally accepted.

# Proposed Solution
We know that most people who are trying to customize `staticwebapp.config.json` should have an idea of what they are doing. Nitro is trying to ensure compatibility and ease of use out of the box which is a good philosophy.

I think a good first pass solution would be to not start with a predefined configuration but to load `staticwebapp.config.json` if the file is present.

I think the properties: `platform` and `navigationFallback` can stay as overwriting what the user has because they are fairly standard and from my observations in this thread and other issues/discussion, no one has needed or tried to customize those.

`routes` seems like the section that is most modified or in my case personally I want to add an additional property `networking` which Nitro doesn't add or really care about.

For routes I propose loading existing `routes` in the `staticwebapp.config.json`  and appending to what currently exists if an object doesn't already exist.

For every other property (anything besides `platform`, `routes`, and `navigationFallback` we should leave it be and keep it included in the configuration file when we load the existing file in the beginning.

**Comment by pi0** [comment]
#1344
